app/components/Showcases/Barter.jsx

```diff
@@ -0,0 +1,724 @@
+                to_account: this.state.to_account.get("id"),
+                amount: sendAmount.getAmount(),
+                asset_id: asset.get("id"),
+                memo: this.state.memo
+                    ? new Buffer(this.state.memo, "utf-8")
+                    : this.state.memo,
+                propose: this.state.from_account, //  propose
+                feeAsset: item.from_feeAsset
+                    ? item.from_feeAsset.get("id")
+                    : "1.3.0"
+            });
+        });
+
+        this.state.to_barter.forEach(item => {
+            const asset = item.to_asset;
+            let amount = item.to_amount;
+            sendAmount = new Asset({
+                real: amount,
+                asset_id: asset.get("id"),
+                precision: asset.get("precision")
+            });
+            transfer_list.push({
+                from_account: this.state.to_account.get("id"),
+                to_account: this.state.from_account.get("id"),
+                amount: sendAmount.getAmount(),
+                asset_id: asset.get("id"),
+                memo: this.state.memo
+                    ? new Buffer(this.state.memo, "utf-8")
+                    : this.state.memo,
+                propose: this.state.to_account, //  propose
+                feeAsset: item.to_feeAsset
+                    ? item.to_feeAsset.get("id")
+                    : "1.3.0"
+            });
+        });
+
+        AccountActions.transfer_list(transfer_list)
+            .then(() => {
+                this.onClose();
+                TransactionConfirmStore.unlisten(this.onTrxIncluded);
+                TransactionConfirmStore.listen(this.onTrxIncluded);
+            })
+            .catch(e => {
+                let msg = e.message
+                    ? e.message.split("\n")[1] || e.message
+                    : null;
+                console.log("error: ", e, msg);
+                this.setState({from_error: msg, to_error: msg});
+            });
+    }
+
+    onTrxIncluded(confirm_store_state) {
+        if (
+            confirm_store_state.included &&
+            confirm_store_state.broadcasted_transaction
+        ) {
+            // this.setState(Transfer.getInitialState());
+            TransactionConfirmStore.unlisten(this.onTrxIncluded);
+            TransactionConfirmStore.reset();
+        } else if (confirm_store_state.closed) {
+            TransactionConfirmStore.unlisten(this.onTrxIncluded);
+            TransactionConfirmStore.reset();
+        }
+    }
+
+    render() {
+        let {
+            from_name,
+            to_name,
+            from_account,
+            to_account,
+            from_balanceError,
+            to_balanceError,
+            from_barter,
+            to_barter,
+            amount_index
+        } = this.state;
+        let {from_asset_types, to_asset_types} = this._getAvailableAssets();
+        let smallScreen = window.innerWidth < 850 ? true : false;
+        let asset;
```

Comment:
`asset` is not filled. In the AmountSelector change the asset, the total amount indicator does not update.